backend/repository/repository.go

```diff
@@ -44,6 +45,7 @@ type Repository struct {
 	// Cached error when getting gateway config.
 	gatewayConfigError error
 	sync.RWMutex
+	validator *codegen.SchemaValidator
```

Comment:
`validator` is not a state for `repository` and it is cheap to create. So it should be introduced as a local variable right before schema validation.